src/test/java/org/codehaus/plexus/archiver/zip/ZipArchiverTest.java

```diff
@@ -889,18 +895,48 @@ public void testForcedFileModes()
     public void testFixedEntryModificationTime()
             throws IOException
     {
-        final long almostMinDosTime = 315532802000L;
+        final long almostMinDosTime = toLocalTimeZone( 315532802000L );
+
         final File zipFile = getTestFile( "target/output/zip-with-fixed-entry-modification-times.zip" );
         final ZipArchiver archiver = getZipArchiver( zipFile );
         archiver.setLastModifiedDate( new Date( almostMinDosTime ) );
         archiver.addDirectory( new File( "src/test/resources/zip-timestamp" ) );
         archiver.createArchive();
 
         assertTrue( zipFile.exists() );
-        final ZipFile zf = new ZipFile( zipFile );
-        assertEquals( almostMinDosTime, zf.getEntry( "file-with-even-time.txt" ).getTime() );
-        assertEquals( almostMinDosTime, zf.getEntry( "file-with-odd-time.txt" ).getTime() );
-        assertEquals( almostMinDosTime, zf.getEntry( "foo/" ).getTime() );
+        try ( final ZipFile zf = new ZipFile( zipFile ) )
+        {
+            assertEquals( almostMinDosTime, zf.getEntry( "file-with-even-time.txt" ).getTime() );
+            assertEquals( almostMinDosTime, zf.getEntry( "file-with-odd-time.txt" ).getTime() );
+            assertEquals( almostMinDosTime, zf.getEntry( "foo/" ).getTime() );
+        }
+    }
+
+    /**
+     * Takes a timestamp located in GMT TZ and convert it to the default system TZ. This makes the test independent of
```

Comment:
a timestamp is not really "located in GMT": if we can find a better explanations, that would help a lot...

proposal: takes a timestamp, turns it into a textual representation based on GMT, then translated it into a timestamp in local timezone.
Such trickery is necessary because:
1. zip binary format does not take timezone into account: 315532802000L is the minimal DOS time "without timezone", which is equivalent to GMT (0 offset)
2. java ZipEntry time implementation transforms such timezone-less value into local timezone: see `java.util.zip.ZipUtils#javaToDosTime`
```Instant instant = Instant.ofEpochMilli(time);
LocalDateTime ldt = LocalDateTime.ofInstant(instant, ZoneId.systemDefault());```